src/broker/boot_config.c

```diff
@@ -141,28 +134,38 @@ static json_t *boot_config_expand_hosts (json_t *hosts)
         return NULL;
     }
     json_array_foreach (hosts, index, value) {
-        struct map_arg maparg = { .hosts = nhosts, .entry = value};
-        const char *host;
+        struct hostlist *hl = NULL;
+        const char *host, *s;
 
         if (json_unpack (value, "{s:s}", "host", &host) < 0) {
             log_msg ("Config file error [bootstrap]: missing host field");
             log_msg ("Hint: hosts entries must be table containing a host key");
             goto error;
         }
-        if (idset_format_map (host, boot_config_map_hosts, &maparg) < 0) {
+        if (!(hl = hostlist_decode (host))) {
             log_err ("Config file error [bootstrap]");
-            log_msg ("Hint: text enclosed by square brackets must be an idset");
             goto error;
         }
+        s = hostlist_first (hl);
+        while (s) {
+            if (boot_config_append_host (nhosts, s, value) < 0) {
+                log_err ("Config file error [bootstrap]");
```

Comment:
Though it looks like the only way `boot_config_append_host()` can fail is due to ENOMEM (I may have missed something). It might still be useful to give more context in this error, e.g.
```suggestion
                log_err ("Config file error [bootstrap]: appending host %s", s);
```